storage/clickhouse/clickhouse.go

```diff
@@ -277,6 +281,54 @@ func (ch *clickHouse) convertReadRequest(request *prompb.Query) base.Query {
 	return q
 }
 
+func (ch *clickHouse) prepareClickHouseQuery(query *prompb.Query, metricName string) (string, error) {
+	var clickHouseQuery string
+	var conditions []string
+	conditions = append(conditions, fmt.Sprintf("metric_name = '%s'", metricName))
+	for _, m := range query.Matchers {
+		switch m.Type {
+		case prompb.LabelMatcher_EQ:
+			conditions = append(conditions, fmt.Sprintf("JSONExtractString(labels, '%s') = '%s'", m.Name, m.Value))
```

Comment:
why are we not using JSON object column rather than string column in filtering?

Reply:
Is this the reason?
```
I had to use labels with JSONExtractString because an user is free to enter whatever they wish in the label key/value but the coulmn might not exist in labels_object JSON and it throws no column error.
```

Reply:
Yes, that's the reason.